### ISSUE TITLE
"diffs didn't match during apply" when updating Scale Set

### ISSUE THREAD

**Issue by shibataka000** [terraform-providers/terraform-provider-azurerm#1739]
### Community Note

* Please vote on this issue by adding a :+1: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.11.7
+ provider.azurerm v1.12.0
```

### Affected Resource(s)

* azurerm_virtual_machine_scale_set

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "sample" {
  name = "sample"
  location = "Japan East"
}

resource "azurerm_virtual_network" "sample" {
  name = "sample"
  resource_group_name = "${azurerm_resource_group.sample.name}"
  location = "${azurerm_resource_group.sample.location}"
  address_space = ["10.0.0.0/16"]
}

resource "azurerm_subnet" "sample" {
  name = "sample"
  resource_group_name = "${azurerm_resource_group.sample.name}"
  virtual_network_name = "${azurerm_virtual_network.sample.name}"
  address_prefix = "10.0.2.0/24"
}

resource "azurerm_virtual_machine_scale_set" "sample" {
  name = "sample"
  resource_group_name = "${azurerm_resource_group.sample.name}"
  location = "${azurerm_resource_group.sample.location}"
  upgrade_policy_mode = "Manual"

  sku {
    name = "Standard_DS1_v2"
    tier = "Standard"
    capacity = 1
  }

  storage_profile_image_reference {
    publisher = "Canonical"
    offer = "UbuntuServer"
    sku = "16.04.0-LTS"
    version = "latest"
  }

  storage_profile_os_disk {
    name = ""
    caching = "ReadWrite"
    create_option = "FromImage"
    managed_disk_type = "Premium_LRS"
  }

  storage_profile_data_disk {
    lun = 0
    caching = "ReadWrite"
    create_option = "Empty"
    disk_size_gb = 10
  }

  os_profile {
    computer_name_prefix = "sample"
    admin_username = "ubuntu"
    admin_password = "password"
  }

  os_profile_linux_config {
    disable_password_authentication = true
    ssh_keys {
      path = "/home/ubuntu/.ssh/authorized_keys"
      key_data = "${file("~/.ssh/azure_default.pub")}"
    }
  }

  network_profile {
    name = "TestNetworkProfile"
    primary = true
    ip_configuration {
      name = "TestIPConfiguration"
      subnet_id = "${azurerm_subnet.sample.id}"
    }
    # network_security_group_id = "${azurerm_network_security_group.sample.id}"
  }
}

# resource "azurerm_network_security_group" "sample" {
#   name = "sample"
#   resource_group_name = "${azurerm_resource_group.sample.name}"
#   location = "${azurerm_resource_group.sample.location}"

#   security_rule {
#     name = "SSH"
#     priority = 1001
#     direction = "Inbound"
#     access = "Allow"
#     protocol = "Tcp"
#     source_port_range = "*"
#     destination_port_range = "22"
#     source_address_prefix = "*"
#     destination_address_prefix = "*"
#   }
# }
```

### Debug Output
https://gist.github.com/shibataka000/872f7336be98a8e4be277fa6933b9779

### Panic Output

n/a

### Expected Behavior

Terraform should have updated the scale set with the network security group.

### Actual Behavior

Got the "diffs didn't match during apply" error shown in the gist.

### Steps to Reproduce

1. `terraform apply`.
2. Revert following statement which is commented out.
    - `resource "azurerm_network_security_group" "sample" { ... }`
    - `network_security_group_id = "${azurerm_network_security_group.sample.id}"`
3. `terraform apply` again.
4. Get the above error.

**Comment by dubeaud** [comment]
We are also having this issue with VMSS extensions on an apply.

**Comment by langemol** [comment]
I too get this error, when trying to add an id to the list azurerm_virtual_machine_scale_set.network_profile.ip_configuration.load_balancer_inbound_nat_rules_ids

```hcl
Terraform v0.11.13
+ provider.azurerm v1.23.0
+ provider.random v2.0.0
```

```hcl
Terraform Version: 0.11.13
    Resource ID: azurerm_virtual_machine_scale_set.vmScaleSet
    Mismatch reason: attribute mismatch: network_profile.121798411.dns_settings.#
    Diff One (usually from plan): *terraform.InstanceDiff{mu:sync.Mutex{state:0, sema:0x0}, Attributes:map[string]*terraform.ResourceAttrDiff{"network_profile.~121798411.ip_configuration.0.load_balancer_backend_address_pool_ids.4058620203":*terraform.ResourceAttrDiff{Old:"", New:"/subscriptions/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/resourceGroups/apiproxyrg/providers/Microsoft.Network/loadBalancers/apiproxylb/backendAddressPools/apiproxybackendpool", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.ip_configuration.0.application_gateway_backend_address_pool_ids.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.dns_settings.0.dns_servers.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.ip_configuration.0.public_ip_address_configuration.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.ip_configuration.0.primary":*terraform.ResourceAttrDiff{Old:"true", New:"false", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.dns_settings.#":*terraform.ResourceAttrDiff{Old:"1", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.network_security_group_id":*terraform.ResourceAttrDiff{Old:"/subscriptions/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/resourceGroups/apiproxyrg/providers/Microsoft.Network/networkSecurityGroups/apiproxynsg", New:"", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.name":*terraform.ResourceAttrDiff{Old:"", New:"apiproxynp", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.ip_configuration.0.subnet_id":*terraform.ResourceAttrDiff{Old:"", New:"/subscriptions/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/resourceGroups/apiproxyrg/providers/Microsoft.Network/virtualNetworks/apiproxyvm/subnets/apiproxysubnet", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.ip_configuration.0.primary":*terraform.ResourceAttrDiff{Old:"", New:"true", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.ip_forwarding":*terraform.ResourceAttrDiff{Old:"", New:"false", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.dns_settings.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.ip_configuration.0.application_security_group_ids.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.ip_configuration.0.subnet_id":*terraform.ResourceAttrDiff{Old:"/subscriptions/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/resourceGroups/apiproxyrg/providers/Microsoft.Network/virtualNetworks/apiproxyvm/subnets/apiproxysubnet", New:"", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.name":*terraform.ResourceAttrDiff{Old:"apiproxynp", New:"", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.ip_configuration.#":*terraform.ResourceAttrDiff{Old:"0", New:"1", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.ip_configuration.0.load_balancer_backend_address_pool_ids.#":*terraform.ResourceAttrDiff{Old:"1", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.primary":*terraform.ResourceAttrDiff{Old:"true", New:"false", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.ip_forwarding":*terraform.ResourceAttrDiff{Old:"false", New:"false", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.primary":*terraform.ResourceAttrDiff{Old:"", New:"true", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.ip_configuration.#":*terraform.ResourceAttrDiff{Old:"1", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.ip_configuration.0.load_balancer_backend_address_pool_ids.4058620203":*terraform.ResourceAttrDiff{Old:"/subscriptions/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/resourceGroups/apiproxyrg/providers/Microsoft.Network/loadBalancers/apiproxylb/backendAddressPools/apiproxybackendpool", New:"", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.accelerated_networking":*terraform.ResourceAttrDiff{Old:"false", New:"false", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.ip_configuration.0.load_balancer_backend_address_pool_ids.#":*terraform.ResourceAttrDiff{Old:"0", New:"1", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.ip_configuration.0.public_ip_address_configuration.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.ip_configuration.0.application_gateway_backend_address_pool_ids.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.ip_configuration.0.load_balancer_inbound_nat_rules_ids.#":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.ip_configuration.0.application_security_group_ids.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.ip_configuration.0.name":*terraform.ResourceAttrDiff{Old:"", New:"apiproxyipconfig", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.network_security_group_id":*terraform.ResourceAttrDiff{Old:"", New:"/subscriptions/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/resourceGroups/apiproxyrg/providers/Microsoft.Network/networkSecurityGroups/apiproxynsg", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.121798411.ip_configuration.0.name":*terraform.ResourceAttrDiff{Old:"apiproxyipconfig", New:"", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~121798411.accelerated_networking":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}}, Destroy:false, DestroyDeposed:false, DestroyTainted:false, Meta:map[string]interface {}(nil)}
    Diff Two (usually from apply): *terraform.InstanceDiff{mu:sync.Mutex{state:0, sema:0x0}, Attributes:map[string]*terraform.ResourceAttrDiff(nil), Destroy:false, DestroyDeposed:false, DestroyTainted:false, Meta:map[string]interface {}(nil)}
```

**Comment by ocastle** [comment]
Encountered this error when adding a public ip address, loadbalancer, with backend pool to an existing scaleset. With the error, the resources were still created.

Terraform v0.11.13

* provider.azuread: version = "~> 0.2"
* provider.template: version = "~> 2.1"

erraform Version: 0.11.13

Resource ID: azurerm_virtual_machine_scale_set.vmScaleSet
Mismatch reason: attribute mismatch: network_profile.2883930349.dns_settings.#

Diff One (usually from plan): *terraform.InstanceDiff{mu:sync.Mutex{state:0, sema:0x0}, Attributes:map[string]*terraform.ResourceAttrDiff{"network_profile.2883930349.primary":*terraform.ResourceAttrDiff{Old:"true", New:"false", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.ip_configuration.#":*terraform.ResourceAttrDiff{Old:"0", New:"1", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.ip_configuration.0.load_balancer_inbound_nat_rules_ids.#":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.network_security_group_id":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.ip_configuration.0.application_security_group_ids.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.name":*terraform.ResourceAttrDiff{Old:"default", New:"", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.ip_forwarding":*terraform.ResourceAttrDiff{Old:"false", New:"false", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.accelerated_networking":*terraform.ResourceAttrDiff{Old:"false", New:"false", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.dns_settings.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.dns_settings.#":*terraform.ResourceAttrDiff{Old:"1", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.ip_configuration.0.load_balancer_backend_address_pool_ids.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.ip_configuration.0.public_ip_address_configuration.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.network_security_group_id":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.ip_configuration.0.public_ip_address_configuration.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.ip_configuration.0.application_security_group_ids.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.ip_configuration.0.name":*terraform.ResourceAttrDiff{Old:"default", New:"", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.ip_configuration.0.subnet_id":*terraform.ResourceAttrDiff{Old:"/subscriptions/****************/resourceGroups/rg-vn-default-01/providers/Microsoft.Network/virtualNetworks/vn-default-01/subnets/scaleset", New:"", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.accelerated_networking":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.ip_configuration.#":*terraform.ResourceAttrDiff{Old:"1", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.ip_configuration.0.primary":*terraform.ResourceAttrDiff{Old:"", New:"true", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.ip_forwarding":*terraform.ResourceAttrDiff{Old:"", New:"false", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.ip_configuration.0.load_balancer_backend_address_pool_ids.#":*terraform.ResourceAttrDiff{Old:"", New:"", NewComputed:true, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.ip_configuration.0.application_gateway_backend_address_pool_ids.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.ip_configuration.0.name":*terraform.ResourceAttrDiff{Old:"", New:"default", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.ip_configuration.0.application_gateway_backend_address_pool_ids.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.ip_configuration.0.subnet_id":*terraform.ResourceAttrDiff{Old:"", New:"/subscriptions/****************/resourceGroups/rg-vn-default-01/providers/Microsoft.Network/virtualNetworks/vn-default-01/subnets/scaleset", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.name":*terraform.ResourceAttrDiff{Old:"", New:"default", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.~2883930349.primary":*terraform.ResourceAttrDiff{Old:"", New:"true", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.dns_settings.0.dns_servers.#":*terraform.ResourceAttrDiff{Old:"0", New:"0", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}, "network_profile.2883930349.ip_configuration.0.primary":*terraform.ResourceAttrDiff{Old:"true", New:"false", NewComputed:false, NewRemoved:true, NewExtra:interface {}(nil), RequiresNew:false, Sensitive:false, Type:0x0}}, Destroy:false, DestroyDeposed:false, DestroyTainted:false, Meta:map[string]interface {}(nil)}

Diff Two (usually from apply): *terraform.InstanceDiff{mu:sync.Mutex{state:0, sema:0x0}, Attributes:map[string]*terraform.ResourceAttrDiff{}, Destroy:false, DestroyDeposed:false, DestroyTainted:false, Meta:map[string]interface {}(nil)}

**Comment by tombuildsstuff** [comment]
hi @shibataka000 @steve-hawkins @dubeaud @langemol @ocastle 

We're currently working on version 2.0 of the Azure Provider which we previously announced in #2807.

As a part of this we're introducing five new resources which will supersede the existing `azurerm_virtual_machine` and `azurerm_virtual_machine_scale_set` resources:

* `azurerm_linux_virtual_machine`
* `azurerm_linux_virtual_machine_scale_set`
* `azurerm_virtual_machine_scale_set_extension`
* `azurerm_windows_virtual_machine`
* `azurerm_windows_virtual_machine_scale_set`

We've recently opened the PR for the first of these new resources #4518 - and I'm able to confirm that this bug is now fixed in the new Virtual Machine Scale Set resource - however unfortunately we have no plans to backport this to the existing `azurerm_virtual_machine_scale_set` resource.

We've posted an update in #2807 with more information on this (we plan to release the new Virtual Machine / Virtual Machine Scale Set resources in a Beta form in a future 1.x version of the Azure Provider and subsequently release them as "GA" in 2.0) - but as we plan to post additional status updates for 2.0 in #2807 I'd suggest subscribing to that issue for updates.

This issue's been assigned to the milestone "2.0" since this is where this will ship - however (due to the way that closing Github Issues from PR's works, to be able to track this back for future users) this issue will be closed once the first of the new resources have been merged.

Thanks!
